src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -31,12 +31,25 @@ internal sealed class BuildServerConnection
         /// </summary>
         internal static bool IsCompilerServerSupported => GetPipeName("") is object;
 
+        /// <summary>
+        /// Create a build request for processing on the server. 
+        /// </summary>
+        /// <remarks>
+        /// Even though compilation itself does not specifically require a temporary directory to be successful
+        /// this API deliberately requires <paramref name="tempDirectory"/> to have a value. The reason for this is
+        /// that the server itself requires a temporary directory to startup and it uses the same logic as the client
+        /// to calculate it.
+        /// 
+        /// That means if the client can't calculate the temporary directory the server won't be able to either and 
+        /// hence won't be able to start. So creating <see cref="BuildRequest"/> instances isn't useful cause it's 
+        /// wasting time trying to start a server that won't be able to start.
+        /// </remarks>
         internal static BuildRequest CreateBuildRequest(
             Guid requestId,
             RequestLanguage language,
             List<string> arguments,
             string workingDirectory,
-            string tempDirectory,
+            string? tempDirectory,
```

Comment:
>? [](http://example.com/codeflow?start=18&length=1)

Is the nullable annotation necessary given that the  method requires a value?

Reply:
Good catch. The annotation is correct here, the `Debug.Assert` is wrong.

Reply:
The doc comment says that this API requires `tempDirectory` to have a value even though it's nullable. Is the comment wrong?